### DISCUSSION TITLE
Create a timestamp locked "vault" for funds in Firefly

### DISCUSSION THREAD

**Discussion by Phyloiota** [iotaledger/firefly#1548] [Features / Enhancements]
## Description

Feature request from user "Stefanoste"

When it comes to crypto the biggest concern is security, we all know sometimes shit can happen  and even with nano  ledger time to time you can feel unsafe 
 
Firefly is fantastic and in terms of security I am sure will be like no one , but why don't enforce it even more and add a kind of "vault" inside ??

The "vault" is a  timestamp lock where you decide to lock some/all funds and be available "X-time " after you requested to  unlock

Let's say I have 10gi , because I don't really need it to be all available in the wallet ,for security , I decide to lock 5gi with the option to make them available again  (two/four/eight or what ever ) hours after the unlock  request 
When you request unlock you get notified 

If someone for whatever reason accesses my wallet can only steal the amount available and not the rest unless of course will wait whatever time need it to unlock "vault" 

## Motivation

Explain why this feature is needed.

## Requirements

Write a list of what you want this feature to do.

1.
2.
3.

## Open questions (optional)

Use this section to ask any questions that are related to the feature.

## Are you planning to do it yourself in a pull request?

Yes/No.

**Comment by Travetown** [comment]
I believe that this could possibly be integrated into Firefly, but then the lock only applies to the Firefly app.  The problem, however, is that anyone with access to the seed can still move the tokens.  There is no lock on the ledger.  Sending cannot be prevented if someone is working with the stolen seed on another computer.  But it may be possible with smart contracts.

**Comment by Stesy73** [comment]
if use smart contract 

> I believe that this could possibly be integrated into Firefly, but then the lock only applies to the Firefly app. The problem, however, is that anyone with access to the seed can still move the tokens. There is no lock on the ledger. Sending cannot be prevented if someone is working with the stolen seed on another computer. But it may be possible with smart contracts.

smart-contract should solve this limitation, so even if you have the seed you will not be able to move funds unless you execute the request first and wait 
Maybe considering to allow "vault" a maximum amount up to 80/90% of the available to avoid running out of money in case of emergency/needs and maximum lock period up to 6/8 hours as a security feature
The notification also plays an important  part in this feature
